sql-ballerina/tests/complex-query-test.bal

```diff
@@ -691,3 +693,31 @@ function testGetArrayTypes4() returns error? {
     };
     test:assertEquals(value, expectedData, "Expected data did not match.");
 }
+
+type UserDefinedTypes record {
+    int udt_int;
+    boolean udt_boolean;
+    string udt_string;
+    float udt_float;
+};
+
+@test:Config {
+    groups: ["query", "query-complex-params"]
+}
+function testUserDefinedTypes() returns error? {
+    MockClient dbClient = check new (url = complexQueryDb, user = user, password = password);
+    stream<record{}, error?> queryResult = dbClient->query("SELECT udt_int, udt_boolean, udt_string, udt_float " +
+    "from UserDefinedTypesTable where udt_int = 1", UserDefinedTypes);
```

Comment:
```suggestion
    stream<UserDefinedTypes, error?> queryResult = dbClient->query("SELECT udt_int, udt_boolean, udt_string, udt_float " +
    "from UserDefinedTypesTable where udt_int = 1");
```

Reply:
shall we infer the type from the assigned stream type as much as we can. as it is the recommended way

Reply:
corrected in https://github.com/ballerina-platform/module-ballerina-sql/pull/291/commits/e27bcd715154e3e8a2ee3f11461dd1ef011c526a